documentation/user-consent.md

```diff
@@ -0,0 +1,220 @@
+# User Consent Management
+
+This document captures guidelines for Operator platform to handle user consents.
+
+## Table of Contents
+
+- [User Consent Management](#user-consent-management)
+  - [Table of Contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [Glossary of Terms and Concepts](#glossary-of-terms-and-concepts)
+  - [Purposes](#purposes)
+    - [Purpose levels](#purpose-levels)
+    - [Purpose relationship with other key concepts](#purpose-relationship-with-other-key-concepts)
+    - [Purpose definition](#purpose-definition)
+    - [Using purpose parameter in the authorization request](#using-purpose-parameter-in-the-authorization-request)
+  - [User Identity](#user-identity)
+  - [User Authentication/Authorization \& Consent Management](#user-authenticationauthorization--consent-management)
+    - [Authentication mechanism/s](#authentication-mechanisms)
+    - [Consent capture and storage](#consent-capture-and-storage)
+    - [Athorization flows / grant types](#athorization-flows--grant-types)
+      - [Authorization code flow](#authorization-code-flow)
+      - [JWT-bearer flow](#jwt-bearer-flow)
+      - [CIBA flow](#ciba-flow)
+      - [Client Credentials flow](#client-credentials-flow)
+      - [... (other grant types)](#-other-grant-types)
+  - [Consent API](#consent-api)
+  - [Annex: Flow source code](#annex-flow-source-code)
+
+
+## Introduction
+
+Operator platform implementing CAMARA should be built with a privacy-by-design approach to fully comply with data protection regulations like the [GDPR regulation](https://gdpr-info.eu/) in Europe, which provides a great level of protection for user's privacy. This means that an API that processes Personal Information needs a user consent. Consents are given by users to legal entities to process personal data under a specific purpose.
```

Comment:
this isn't true - an API that processes Personal Information needs a 'legal basis' for doing so, one of which would be user consent.

Reply:
Yes, I agree. We will update the document to rephrase it correctly. As explained above, we were using user consent as a generic term and not specifically linked to the "consent" legal basis. But it does seem to be causing some confusion, so we will try to change the doc accordingly.
If you have any suggestions to do so, please don't hesitate to suggest a change.

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/13/commits/4b761353f33ee263e98e8309c8fb698908401206